webdriver/tests/bidi/network/continue_with_auth/invalid.py

```diff
@@ -0,0 +1,95 @@
+import pytest
+import webdriver.bidi.error as error
+
+from .. import PAGE_EMPTY_TEXT, RESPONSE_COMPLETED_EVENT
+
+pytestmark = pytest.mark.asyncio
+
+
+@pytest.mark.parametrize("value", ["beforeRequestSent", "responseStarted"])
+async def test_params_request_invalid_phase(setup_blocked_request, bidi_session, value):
+    request = await setup_blocked_request(value)
+
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.network.continue_with_auth(request=request, action="cancel")
+
+
+@pytest.mark.parametrize("value", [None, False, 42, {}, []])
+async def test_params_request_invalid_type(bidi_session, value):
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.network.continue_with_auth(request=value, action="cancel")
+
+
+@pytest.mark.parametrize("value", ["", "foo"])
+async def test_params_request_invalid_value(bidi_session, value):
+    with pytest.raises(error.NoSuchRequestException):
+        await bidi_session.network.continue_with_auth(request=value, action="cancel")
+
+
+async def test_params_request_no_such_request(
+    bidi_session, setup_network_test, wait_for_event, fetch, url
+):
+    await setup_network_test(
+        events=[
+            RESPONSE_COMPLETED_EVENT,
+        ]
+    )
+    on_response_completed = wait_for_event(RESPONSE_COMPLETED_EVENT)
+
+    text_url = url(PAGE_EMPTY_TEXT)
+    await fetch(text_url)
+
+    response_completed_event = await on_response_completed
+    request = response_completed_event["request"]["request"]
+
+    with pytest.raises(error.NoSuchRequestException):
+        await bidi_session.network.continue_with_auth(request=request, action="cancel")
+
+
+@pytest.mark.parametrize("value", [None, False, 42, {}, []])
+async def test_params_action_invalid_type(setup_blocked_request, bidi_session, value):
+    request = await setup_blocked_request("authRequired")
+
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.network.continue_with_auth(request=request, action=value)
+
+
+@pytest.mark.parametrize("value", ["", "foo"])
+async def test_params_action_invalid_value(setup_blocked_request, bidi_session, value):
+    request = await setup_blocked_request("authRequired")
+
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.network.continue_with_auth(request=request, action=value)
+
+
+@pytest.mark.parametrize(
+    "value",
+    [
+        {"type": "password", "password": "foo"},
+        {"type": "password", "username": "foo"},
+        {
+            "type": "password",
```

Comment:
We should also have a test with `username` and `password` but an invalid `type` value.

Reply:
reasonPhrase accepts any kind of string so I can't think of an invalid value to test here?

Reply:
Done, also tested invalid type values

Reply:
nit: I don't the invalid type is tested